internal/core/item_rendering.rs

```diff
@@ -286,12 +286,26 @@ pub trait RenderImage {
     fn tiling(self: Pin<&Self>) -> (ImageTiling, ImageTiling);
 }
 
+/// Trait for an item has font properties
+#[allow(missing_docs)]
+pub trait HasFont {
+    fn font_family(self: Pin<&Self>) -> SharedString;
+    fn font_weight(self: Pin<&Self>) -> i32;
+    fn font_size(self: Pin<&Self>) -> LogicalLength;
+    fn letter_spacing(self: Pin<&Self>) -> LogicalLength;
+    fn font_italic(self: Pin<&Self>) -> bool;
+}
```

Comment:
I'm working my way through reviewing this. Perhaps this trait could return some kind of `FontInfo` struct instead to reduce the number of separate functions that need to be implemented?

Reply:
Yep, similar to what Olivier pointed out. I'll just have `font_request` here, I think.